src/WebJobs.Script/Host/ScriptHost.cs

```diff
@@ -539,6 +541,9 @@ protected virtual void Initialize()
                     _blobLeaseManager.HasLeaseChanged += BlobLeaseManagerHasLeaseChanged;
                 }
 
+                //Apply to JobHostMetadata
+                ApplyJobHostMetadata(functionMetadata);
```

Comment:
I think you might have an ordering issue here. This method will be using the metadata provider which requires the core SDK to have indexed all the functions. To index the functions, it uses the TypeLocator that we create **below** which will contain the set of types to index.

Have you debugged through this to make sure this is all working?

Reply:
Indeed. After debugging I found the ordering issue. I moved ApplyJobHostMetadata after SDK have index for all function (StartAsync). Good catch!